os-fpga/FOEDAG
# [RG-103] Close button warning messagebox added in main window

> ### Motivate of the pull request
> - [ ] **Currently when user closes the project knowingly or unknowingly ,  GUI just closes the project and does not ask user if user really wants to close the project.**

> #### What is currently done? (Provide issue link if applicable)
A warning message box is added to close window event when user wants to close the GUI. It will prompt a warning dialog and ask "Do you really want to close the project". It would help the user to avoid close action by accidentally clicked close button. 
>
> #### What does this pull request change?
An user should be warn or should be asked “do you really wants to close the project”. 

**Testing Steps:**

- Run the GUI
- Close the GUI by clicking on "x" button on top right corner
- Warning dialog will open with to options "Ok" and "Cancel".
- Verify Ok button should close the GUI and Cancel should cancel the action of close button.

**Screenshot**
Expected Output:
![image](https://user-images.githubusercontent.com/109368108/192502670-27d1807e-a1c5-40bd-97a3-d3db45b810c8.png)


> ### Which part of the code base require a change
> 
> - [ ] Library: <Specify the library name>
> - [ ] Plug-in: <Specify the plugin name>
> - [ ] Engine
> - [ ] Documentation
> - [ ] Regression tests
> - [ ] Continous Integration (CI) scripts

> ### Impact of the pull request

> - [ ] Require a change on Quality of Results (QoR)
> - [ ] Break back-compatibility. If so, please list who may be influenced.
